geonode/layers/download_handler.py

```diff
@@ -51,9 +51,35 @@ def get_download_response(self):
         that allow the resource download
         """
         resource = self.get_resource()
+        if not resource:
+            raise Http404("Resource requested is not available")
         response = self.process_dowload(resource)
         return response
 
+    @property
+    def is_ajax_safe(self):
+        """
+        We use that USE_GEOSERVER boolean to decide if use the default
+        WPS download method or use the fallback on the proxy.
+        This means that by default if geoserver is used, the WPS
+        is used
+        """
+        return settings.USE_GEOSERVER
```

Comment:
@mattiagiupponi this assumption is wrong. AJAX is safe if we actually use WPS, but in case an "original" link is set we cannot ensure AJAX, since it could point to an external URL and the client should redirect the user to that URL.